src/users/users.service.ts

```diff
@@ -80,4 +80,32 @@ export class UsersService {
 
     return userProfile;
   }
+
+  async updateUser(id: number, updateUserDto: UpdateUserDto) {
+    const user = await this.usersRepository.findOneOrFail(id);
+    const { firstName, lastName, gender } = updateUserDto;
+
+    this.usersRepository.assign(user, { firstName, lastName, gender });
+
+    if (user.role === EUserRole.TEACHER) {
+      const { teacherInput } = updateUserDto;
+
+      this.entityManager.assign(user.teacher, teacherInput);
```

Comment:
```suggestion
      this.usersRepository.assign(user, { firstName, lastName, gender, teacher: teacherInput });
```

Can't we do this? 
Same comment for assigning values to user.student

Reply:
No bhai. I causes an error which is teacher field contains id but we are giving objects to assign. We can't assign the teacher object with the EM of usersRepository.